diff --git a/node_modules/electrum-client/lib/client.js b/node_modules/electrum-client/lib/client.js
index f068f0b..c3ccd82 100644
--- a/node_modules/electrum-client/lib/client.js
+++ b/node_modules/electrum-client/lib/client.js
@@ -42,7 +42,7 @@ class Client {
           throw new Error('tls package could not be loaded');
         }
         this.connUnsecure = new this.net.Socket();
-        this.conn = new this.tls.TLSSocket(this.connUnsecure, { rejectUnauthorized: false });
+        this.conn = new this.tls.TLSSocket(this.connUnsecure, { requestCert: true, rejectUnauthorized: true });
         break;
       default:
         throw new Error('unknown protocol');
