diff --git a/node_modules/expo-file-system/ios/EXFileSystem/EXFileSystem.m b/node_modules/expo-file-system/ios/EXFileSystem/EXFileSystem.m
index 1877d38..525562d 100644
--- a/node_modules/expo-file-system/ios/EXFileSystem/EXFileSystem.m
+++ b/node_modules/expo-file-system/ios/EXFileSystem/EXFileSystem.m
@@ -772,30 +772,6 @@ - (NSURLSessionUploadTask * _Nullable)createUploadTask:(NSString *)urlString
   resolve([NSNull null]);
 }
 
-EX_EXPORT_METHOD_AS(getFreeDiskStorageAsync, getFreeDiskStorageAsyncWithResolver:(EXPromiseResolveBlock)resolve rejecter:(EXPromiseRejectBlock)reject)
-{
-  NSError *error = nil;
-  NSNumber *freeDiskStorage = [self freeDiskStorageWithError:&error];
-
-  if(!freeDiskStorage || error) {
-    reject(@"ERR_FILESYSTEM_CANNOT_DETERMINE_DISK_CAPACITY", @"Unable to determine free disk storage capacity", error);
-  } else {
-    resolve(freeDiskStorage);
-  }
-}
-
-EX_EXPORT_METHOD_AS(getTotalDiskCapacityAsync, getTotalDiskCapacityAsyncWithResolver:(EXPromiseResolveBlock)resolve rejecter:(EXPromiseRejectBlock)reject)
-{
-  NSError *error = nil;
-  NSNumber *diskCapacity = [self totalDiskCapacityWithError:&error];
-
-  if (!diskCapacity || error) {
-    reject(@"ERR_FILESYSTEM_CANNOT_DETERMINE_DISK_CAPACITY", @"Unable to determine total disk capacity", error);
-  } else {
-    resolve(diskCapacity);
-  }
-}
-
 #pragma mark - Internal methods
 
 - (EXFileSystemUploadType)_getUploadTypeFrom:(NSNumber * _Nullable)type
@@ -1057,22 +1033,4 @@ + (NSString *)generatePathInDirectory:(NSString *)directory withExtension:(NSStr
   return [directory stringByAppendingPathComponent:fileName];
 }
 
-// '<ARCType> *__autoreleasing*' problem solution: https://stackoverflow.com/a/8862061/4337317
-- (NSNumber *)totalDiskCapacityWithError:(out NSError * __autoreleasing *)error
-{
-  NSDictionary *results = [self documentFileResourcesForKeys:@[NSURLVolumeTotalCapacityKey]
-                                                       error:error];
-
-  return results[NSURLVolumeTotalCapacityKey];
-}
-
-// '<ARCType> *__autoreleasing*' problem solution: https://stackoverflow.com/a/8862061/4337317
-- (NSNumber *)freeDiskStorageWithError:(out NSError * __autoreleasing *)error
-{
-  NSDictionary *results = [self documentFileResourcesForKeys:@[NSURLVolumeAvailableCapacityForImportantUsageKey]
-                                                       error:error];
-
-  return results[NSURLVolumeAvailableCapacityForImportantUsageKey];
-}
-
 @end
diff --git a/node_modules/expo-file-system/src/types.ts b/node_modules/expo-file-system/src/types.ts
index 775b4f4..22f2bbe 100644
--- a/node_modules/expo-file-system/src/types.ts
+++ b/node_modules/expo-file-system/src/types.ts
@@ -19,8 +19,6 @@ export interface ExponentFileSystemModule {
   readonly downloadResumableStartAsync?: PlatformMethod;
   readonly downloadResumablePauseAsync?: PlatformMethod;
   readonly getContentUriAsync?: PlatformMethod;
-  readonly getFreeDiskStorageAsync?: PlatformMethod;
-  readonly getTotalDiskCapacityAsync?: PlatformMethod;
   readonly requestDirectoryPermissionsAsync?: PlatformMethod;
   readonly readSAFDirectoryAsync?: PlatformMethod;
   readonly makeSAFDirectoryAsync?: PlatformMethod;
